app/utils/createEntityReducer.js

```diff
@@ -104,6 +104,27 @@ export function optimistic(mutateType?: ?AsyncActionType) {
   };
 }
 
+export function paginationReducer(key: string, fetchType?: ?AsyncActionType) {
+  return (state: any, action: any) => {
+    if (action.type !== get(fetchType, 'SUCCESS')) {
+      return state;
+    }
+
+    if (!action.payload) {
+      return state;
+    }
+
+    return {
+      ...state,
+      hasMore: typeof action.payload.next === 'string',
+      pagination: {
+        next: parse(
+          action.payload.next ? action.payload.next.split('?')[1] : ''
```

Comment:
move this out to a few variables maybe? no need to parse an empty string for example